specs/language/resources.tex

```diff
@@ -0,0 +1,712 @@
+gets converted to the destination type.
+
+\Sec{Typed Buffers}{Resources.tybufs}
+
+\begin{HLSL}
+template <typename T = float4>
+ class Buffer {
+ public:
+   Buffer();
+   Buffer(Buffer buf);
+   Buffer operator=(Buffer buf);
+
+   void GetDimensions(out uint width) const;
+
+   // Element access.
+   T operator[](uint pos) const;
+   T Load(in int pos) const;
+   T Load(in int pos, out uint status) const;
+};
+
+
+template <typename T>
+ class RWBuffer : public Buffer {
+ public:
+   RWBuffer();
+   RWBuffer(RWBuffer buf);
+   RWBuffer operator=(RWBuffer buf);
+
+   // element access
+   T &operator[](uint n);
+};
+}
+\end{HLSL}
+
+\Sub{Constructors}{Resources.tybufs.ctrs}
+
+Read-only Buffers can be defined with explicit or implicit template parameter types.
+\begin{HLSL}
+  Buffer buf1;
+  Buffer<> buf2;
+  Buffer<float4> buf3;
+\end{HLSL}
+Since \texttt{float4} is the default type, all of these definitions are equivalent.
+
+RWBuffers must be defined with explicit template parameter types.
+\begin{HLSL}
+  RWBuffer<float4> buf;
+\end{HLSL}
+
+When defined at global scope, typed buffers are bound to externally-defined backing stores
+using the explicit binding location if provided or the implicit binding if not.
+
+When defined at local scope, typed buffers represent local references
+that can be associated with global buffers when assigned,
+but must be resolvable to a unique global buffer.
+
+\begin{HLSL}
+  Buffer<int> grobuf;
+  RWBuffer<int> grwbuf;
+  void main() {
+    Buffer lrobuf<int> = grobuf;
+    RWBuffer lrwbuf<int> = grwbuf;
+  }
+\end{HLSL}
+Buffer operands to the assignment operator must have the same element type.
+
+\Sub{Dimensions}{Resources.tybufs.dims}
+
+The size of a typed buffer can be retrieved using the \texttt{GetDimensions} method.
+\begin{HLSL}
+void GetDimensions(out uint width) const;
+\end{HLSL}
+
+This returns the full legnth, in elements of the buffer through the \texttt{width} out parameter.
+
+\Sub{Element Access}{Resources.tybufs.access}
+
+The contents of typed buffers can be retrieved using the \texttt{Load} methods
+or the subscript operator.
+
+\begin{HLSL}
+ T Load(in int pos) const;
+ T operator[](uint pos) const;
+\end{HLSL}
+
+These each return the element at the given position \texttt{pos} of the type specified in the buffer definition.
+
+An additional \texttt{Load} method returns the indicated element just as the other,
+but also takes a second \texttt{status} parameter that returns information about the accessed resource.
+\begin{HLSL}
+ T Load(in int pos, out uint status) const;
+\end{HLSL}
+
+The \texttt{status} parameter returns an indication of whether the value retrieved accessed a mapped tile
+resource. This parameter is interpretted using the \texttt{CheckAccessFullyMapped}
+built-in function. If tiled resources are enabled, it returns true if the value was accessed from a mapped
+tile resource and false otherwise. If tiled resourced are not enabled, it will return true.
```

Comment:
Here's my attempt at this:

```suggestion
The \texttt{status} parameter returns an indication of whether all of the retrieved value came from fully mapped parts of the resource.  This parameter must be passed to the built-in function \texttt{CheckAccessFullyMapped} in order to interpret it.  This function returns true is the value was accessed from a fully committed resource, or from fully mapped pages of a sparse resource. If any part of the return value was from unmapped memory then false is returned.
```

Reply:
I took much of this text, but ended up splitting out the `CheckAccessFullyMapped` built-in to its own section where I could go into slightly more depth with less disruption to the explanation of Loads.